src/firewall/core/fw_ifcfg.py

```diff
@@ -54,7 +60,11 @@ def search_ifcfg_of_interface(interface):
             continue
         ifcfg_file = ifcfg("%s/%s" % (config.IFCFGDIR, filename))
         ifcfg_file.read()
-        if ifcfg_file.get("DEVICE") == interface:
+        # on SUSE, ETHERDEVICE is only applicable when creating VLANs so it may not always
+        # be there.
+        if ifcfg_file.get("DEVICE") == interface or \
+           ifcfg_file.get("ETHERDEVICE") == interface or \
+           (suseifcfg and ifcfg_file.get("ETHERDEVICE") is None):
             return ifcfg_file
 
```

Comment:
I think the best way to deal with this is to do yet another search at the end of this file if we haven't yet found a match. This time we just use ifcfg-\<interface\> if it exists without checking for DEVICE. This would let SUSE ifcfg style scripts work without special casing it.

Reply:
Would it make sense to stop looking for `DEVICE` as well? My understanding is that on RedHat you normally have `ifcfg-$interface` and `DEVICE=$interface`. Is there a case in which these two are different? According to the RedHat doc https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/6/html/deployment_guide/s1-networkscripts-interfaces it seems that it always matches the network interface that is being configured.